js/tuxexchange.js

```diff
@@ -0,0 +1,533 @@
+        };
+        const result = await this.privatePost (this.extend (cancelRequest, params));
+        return { 'info': result };
+    }
+
+    parseOrder (order) {
+        const symbol = this.findSymbol (this.safeString (order, 'market_pair'));
+        const orderDate = this.parseDate (this.safeString (order, 'date'));
+        const timestamp = orderDate.valueOf ();
+        const amount = this.asFloat (this.safeString (order, 'amount'));
+        const filled_amount = this.asFloat (this.safeString (order, 'filledamount'));
+        return {
+            'info': order,
+            'id': this.safeString (order, 'id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': 'limit',
+            'side': this.safeString (order, 'type'),
+            'price': this.asFloat (this.safeString (order, 'price')),
+            'amount': amount,
+            'remaining': amount - filled_amount,
+            'filled': filled_amount,
+            'status': 'open',
+            'lastTradeTimestamp': undefined,
+        };
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const openOrdersMap = await this.privatePost (this.extend ({ 'method': 'getmyopenorders' }, params));
+        const openOrders = this.values (openOrdersMap);
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const result = this.parseOrders (openOrders, market, since, limit);
+        return result;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        const myTrades = await this.privatePost (this.extend ({ 'method': 'getmytradehistory' }, params));
+        const results = this.parseTrades (myTrades, symbol, since, limit);
+        return results;
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        if (code === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchDepositAddress () requires a code argument');
+        }
+        await this.loadMarkets ();
+        const addressesData = await this.privatePost (this.extend ({ 'method': 'getmyaddresses' }, params));
+        const addresses = addressesData['addresses'];
+        const addressForCode = addresses[this.currencyId (code)];
+        return {
+            'currency': code,
+            'address': this.checkAddress (addressForCode),
+            'info': addressForCode,
+        };
+    }
+
+    parseTransaction (transaction, type) {
+        const transactionDate = this.parseDate (this.safeString (transaction, 'date'));
+        const timestamp = transactionDate.valueOf ();
+        const marketStatus = this.safeString (transaction, 'status');
+        const status = marketStatus === 'success' ? 'ok' : marketStatus;
```

Comment:
We do this with a unified method `parseTransactionStatus`, see how it is done here: https://github.com/ccxt/ccxt/blob/master/js/upbit.js#L998. Same is true for `parseOrderStatus`.